src/api_server/src/lib.rs

```diff
@@ -67,6 +70,9 @@ pub struct ApiServer {
     /// FD on which we notify the VMM that we have sent at least one
     /// `VmmRequest`.
     to_vmm_fd: EventFd,
+    /// If this flag is set with some value, the process encountered
+    /// a fatal error and it is going to exit.
+    encountered_fatal_error: RefCell<Option<u8>>,
```

Comment:
You should save the error so you can log it before exiting. Also, ideally avoid `RefCell` unless necessary.

```suggestion
    fatal_error_encountered: Option<VmmActionError>,
```

Reply:
Will do. The reason for RefCell usage is explained on a comment below.